### ISSUE TITLE
Extend VarCouldBeVal Rule to Include Class/Object Properties

### ISSUE THREAD

**Issue by tylerbwong** [detekt/detekt#1330]
## Expected Behavior of the rule

Currently, the `VarCouldBeVal` rule only reports on locally-scoped function variables. It would be nice if the rule could be extended to include private top-level `class`/`object` properties as well.

For example

```kt
class Test {
    private var viewModel = BaseViewModel() // Not re-assigned anywhere else in the class
    ...
}
```
should report a violation.

## Context

Theses are valid errors that should be reported to fix unnecessary mutability.

**Comment by tylerbwong** [comment]
I'll actually go ahead and give this one a shot 👍

**Comment by schalkms** [comment]
@tylerbwong do you any update on this feature? Do you need assistance?

**Comment by tylerbwong** [comment]
Hey @schalkms sorry for the late reply. I've actually haven't found the time to work on this lately so it'd probably be best to let someone else work on it 😞

**Comment by schalkms** [comment]
@tylerbwong no worries.. thanks for the reply.

**Comment by BraisGabin** [comment]
This was implemented on 1.20 🎉
